frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/openai/widgets/auto_completion_node_widget.dart

```diff
@@ -280,6 +299,88 @@ class _AutoCompletionInputState extends State<_AutoCompletionInput> {
     });
   }
 
+  Future<void> _onRewrite() async {
+    String previousOutput = _getPreviousOutput();
+    final loading = Loading(context);
+    loading.start();
+    // clear previous response
+    final selection =
+        widget.node.attributes[kAutoCompletionInputStartSelection];
+    if (selection != null) {
+      final start = Selection.fromJson(json.decode(selection)).start.path;
+      final end = widget.node.previous?.path;
+      if (end != null) {
+        final transaction = widget.editorState.transaction;
+        transaction.deleteNodesAtPath(
+          start,
+          end.last - start.last + 1,
+        );
+        await widget.editorState.apply(transaction);
+      }
+    }
+    // generate new response
+    final result = await UserBackendService.getCurrentUserProfile();
+    result.fold((userProfile) async {
+      final openAIRepository = HttpOpenAIRepository(
+        client: http.Client(),
+        apiKey: userProfile.openaiKey,
+      );
+      await openAIRepository.getStreamedCompletions(
+        prompt: _rewritePrompt(previousOutput),
+        temperature: _getRandomTemperature(),
+        onStart: () async {
+          loading.stop();
+          await _makeSurePreviousNodeIsEmptyTextNode();
+        },
+        onProcess: (response) async {
+          if (response.choices.isNotEmpty) {
+            final text = response.choices.first.text;
+            await widget.editorState.autoInsertText(
+              text,
+              inputType: TextRobotInputType.word,
+              delay: Duration.zero,
+            );
+          }
+        },
+        onEnd: () async {},
+        onError: (error) async {
+          loading.stop();
+          await _showError(error.message);
+        },
+      );
+      await _updateGenerationCount();
+    }, (error) async {
+      loading.stop();
+      await _showError(
+        LocaleKeys.document_plugins_autoGeneratorCantGetOpenAIKey.tr(),
+      );
+    });
+  }
+
+  _getPreviousOutput() {
+    final selection =
+        widget.node.attributes[kAutoCompletionInputStartSelection];
+    if (selection != null) {
+      final start = Selection.fromJson(json.decode(selection)).start.path;
+      final end = widget.node.previous?.path;
+      if (end != null) {
+        String lastOutput = "";
+        for (var i = start.last; i < end.last - start.last + 2; i++) {
+          TextNode? textNode =
+              widget.editorState.document.nodeAtPath([i]) as TextNode?;
+          lastOutput = "$lastOutput ${textNode!.toPlainText()}";
+        }
+        return lastOutput.trim();
+      }
+    }
```

Comment:
```dart
final selection =
        widget.editorState.service.selectionService.currentSelection.value;
    if (selection == null) {
      return;
    }
    final textNodes = widget
        .editorState.service.selectionService.currentSelectedNodes
        .whereType<TextNode>()
        .toList(growable: false);
    final input = widget.editorState.getTextInSelection(
      textNodes.normalized,
      selection.normalized,
    );
```

Reply:
use this function `getTextInSelection` to get the selected content.

Reply:
I used the function you are refering but the content which AI is writing is not being selected here in the current situation. That function was returning null, That why I had to get the content by getting the start and end path.